pkg/generate.go

```diff
@@ -71,60 +73,55 @@ func Generate(fs afero.Afero, gitOpsFolder string, outputFolder string, componen
 	}
 
 	var service *corev1.Service
-	var route *routev1.Route
 
-	if len(component.KubernetesResources.Services) == 0 && component.TargetPort != 0 {
+	if len(options.KubernetesResources.Services) == 0 && options.TargetPort != 0 {
 		// If service was not provided, generate a service only if target port was provided
 		// If service was not provided and target port is 0, skip generation
-		service = generateService(component)
-	} else if len(component.KubernetesResources.Services) > 0 {
+		service = generateService(options)
+	} else if len(options.KubernetesResources.Services) > 0 {
 		// If a service was provided, get the first and append the rest to others
-		service, component.KubernetesResources.Services = &component.KubernetesResources.Services[0], component.KubernetesResources.Services[1:]
+		service, options.KubernetesResources.Services = &options.KubernetesResources.Services[0], options.KubernetesResources.Services[1:]
 
 		var otherServices []interface{}
-		for _, service := range component.KubernetesResources.Services {
+		for _, service := range options.KubernetesResources.Services {
 			otherServices = append(otherServices, service)
 		}
 
-		component.KubernetesResources.Others = append(component.KubernetesResources.Others, otherServices...)
+		options.KubernetesResources.Others = append(options.KubernetesResources.Others, otherServices...)
 	}
 
-	if len(component.KubernetesResources.Routes) == 0 && component.TargetPort != 0 {
-		// If route was not provided, generate a route only if target port was provided
-		// If route was not provided and target port is 0, skip generation
-		route = generateRoute(component)
-	} else if len(component.KubernetesResources.Routes) > 0 {
-		// If a route was provided, get the first and append the rest to others
-		route, component.KubernetesResources.Routes = &component.KubernetesResources.Routes[0], component.KubernetesResources.Routes[1:]
+	if len(options.KubernetesResources.Routes) > 0 {
```

Comment:
Are we changing the behaviour of the Generate method (particularly, those that call this Generate function), because before, this function would generate the route if there are no routes provided and the target port is not 0. (See original line 92). Now this code is moved over to the GenerateOverlays method (See line 207).

Just wanted to check that this is intentional.